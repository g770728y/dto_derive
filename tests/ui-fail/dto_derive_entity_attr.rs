extern crate dto_derive;
use dto_derive::Dto;

#[derive(Debug, PartialEq, Dto)]
#[dto(entity = 123)] //~ ERROR expected string literal containing entity type
struct Dto1 { }

#[derive(Debug, PartialEq, Dto)]
#[dto(entity = "Entity")]
#[dto(entity = "Entity")] //~ ERROR already set an entity attribute
struct Dto2 { }

fn main() { }
