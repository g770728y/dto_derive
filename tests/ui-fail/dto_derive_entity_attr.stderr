error: expected string literal containing entity type
 --> $DIR/dto_derive_entity_attr.rs:5:16
  |
5 | #[dto(entity = 123)] //~ ERROR expected string literal containing entity type
  |                ^^^

error: already set an entity attribute
  --> $DIR/dto_derive_entity_attr.rs:10:6
   |
10 | #[dto(entity = "Entity")] //~ ERROR already set an entity attribute
   |      ^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

